deployment/data-streams/changeset/jd_register_nodes.go

```diff
@@ -13,27 +12,65 @@ import (
 type RegisterNodesInput struct {
 	EnvLabel    string
 	ProductName string
-	DONs        DONConfigMap
+	DONs        DONConfigMap `json:"dons,omitempty"`
+	DONsList    []DONConfig  `json:"dons_list,omitempty"`
 }
 
 type DONConfigMap map[string]DONConfig
 
 type DONConfig struct {
+	ID                 int       `json:"id"`
+	ChainSelector      string    `json:"chainSelector"`
 	Name               string    `json:"name"`
 	ChannelConfigStore string    `json:"channelConfigStore"`
 	Verifier           string    `json:"verifier"`
 	Configurator       string    `json:"configurator"`
 	Nodes              []NodeCfg `json:"nodes"`
+	BootstrapNodes     []NodeCfg `json:"bootstrapNodes"`
 }
 
 type NodeCfg struct {
 	Name        string `json:"name"`
 	CSAKey      string `json:"csa_key"`
-	IsBootstrap bool   `json:"isBootstrap"`
+	IsBootstrap bool   `json:"isBootstrap,omitempty"`
+}
+
+func validateNodeSlice(nodes []NodeCfg, nodeType string, donIndex int) error {
+	for _, node := range nodes {
+		if node.Name == "" {
+			return fmt.Errorf("DON[%d] has %s node with empty Name", donIndex, nodeType)
+		}
+		if node.CSAKey == "" {
+			return fmt.Errorf("DON[%d] %s node %s has empty CSAKey", donIndex, nodeType, node.Name)
+		}
+	}
+	return nil
+}
+
+func registerNodesForDON(e deployment.Environment, nodes []NodeCfg, baseLabels []*ptypes.Label, isBootstrap bool) {
```

Comment:
```suggestion

type NodeType int

const (
	NodeTypeOracle NodeType = iota
	NodeTypeBootstrap
)

func registerNodesForDON(e deployment.Environment, nodes []NodeCfg, baseLabels []*ptypes.Label, nodeType NodeType) {
```

Consider using an enum instead of a boolean to distinguish between bootstrap and OCR nodes. This would improve readability, particularly on lines 87 and 88.

Reply:
That makes sense! Just added